SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -211,6 +214,17 @@ public void checkForDecomissions() {
 
     for (SingularityAgent agent : agents.keySet()) {
       boolean foundTask = false;
+      List<SingularityMachineStateHistoryUpdate> history = agentManager.getHistory(
+        agent.getId()
+      );
+
+      // TODO: Is there a guaranteed order these come out of ZK in to prevent sort?
+      //       Probably nbd because this list will be very short, but still
```

Comment:
@ssalinas By any chance do you know this off the top of your head?

Reply:
will want to sort them here, I don't believe our agentManager provides any ordering